### PR TITLE
LPS-174314 Auto SF, use passed in variable

### PR DESCRIPTION
**Pull request by ling-alan-huang**

### CODE REVIEW

**Comment by ling-alan-huang** [comment]
ci:test:relevant

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:test:relevant

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
@austinchiang Can you help check test failures.

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by ling-alan-huang** [comment]
ci:forward

**Comment by austinchiang** [comment]
These tests might be related since they've only been failing on this (and the previous https://github.com/ling-alan-huang/liferay-portal/pull/1682) pull:

- https://testray.liferay.com/home/-/testray/case_results/2232556163
- https://testray.liferay.com/home/-/testray/case_results/2232555959
- https://testray.liferay.com/home/-/testray/case_results/2232556321

These Poshi tests are also failing consistently in this PR: https://testray.liferay.com/home/-/testray/case_results/2232552686, https://testray.liferay.com/home/-/testray/case_results/2232555733 @tim-cao can you please take a look at these? I think you have more knowledge about these tests than me

**Comment by Tim-Cao** [comment]
Hi @ling-alan-huang, there is a regression issue caught by https://testray.liferay.com/home/-/testray/case_results/2232555733
Step to reproduces:

1. Applications Menus > Control Panel > System Settings > Assets > Categorization
2. Change the Maximum Number of Categories per Vocabulary to 3
3. Add a new site
4. Product Menu > Categorization > Categories
5. Add a new vocabulary
6. You will see this message **The maximum number of categories per vocabulary is 3.**
7. Add three categories under this vocabulary
8. Click the ellipsis button of one category

**Expected Results:**
 The Add Subcategory action shouldn't be shown.
 
 **Actual Results:**
  The Add Subcategory action is still shown.

**Comment by ling-alan-huang** [comment]
Thanks @austinchiang @Tim-Cao 
@qz1010 Please check it again, are the failures related [1da3715](https://github.com/ling-alan-huang/liferay-portal/pull/1691/commits/1da371599c41c80e396674a9c7aa4d6177497be0)?

File: modules/apps/asset/asset-categories-admin-web/src/main/java/com/liferay/asset/categories/admin/web/internal/servlet/taglib/util/AssetCategoryActionDropdownItemsProvider.java [hunk 1]
```diff
@@ -64,16 +64,15 @@ public AssetCategoryActionDropdownItemsProvider(
 		_httpServletRequest = httpServletRequest;
 		_renderResponse = renderResponse;
 
-		_themeDisplay = (ThemeDisplay)httpServletRequest.getAttribute(
-			WebKeys.THEME_DISPLAY);
-
 		_assetCategoriesLimitExceeded = _isAssetCategoriesLimitExceeded(
```
Comment (qz1010) [inline modules/apps/asset/asset-categories-admin-web/src/main/java/com/liferay/asset/categories/admin/web/internal/servlet/taglib/util/AssetCategoryActionDropdownItemsProvider.java:67]: In "_isAssetCategoriesLimitExceeded" method uses "_themeDisplay"

**Comment by qz1010** [comment]
see #1723

**Pull request closed without merging**